contracts/MechFactoryBasic.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {AgentMech} from "./AgentMech.sol";
+
+/// @title Mech Factory Basic - Periphery smart contract for managing basic mech creation
+contract MechFactoryBasic {
+    event CreateBasicMech(address indexed mech, uint256 indexed serviceId, uint256 indexed price);
+
+    // Agent factory version number
+    string public constant VERSION = "0.1.0";
+
+    /// @dev Registers service as a mech.
+    /// @param mechMarketplace Mech marketplace address.
+    /// @param serviceRegistry Service registry address.
+    /// @param serviceId Service id.
+    /// @param payload Mech creation payload.
+    /// @return mech The created mech instance address.
+    function createMech(
```

Comment:
Basically mech creation function as before, just depends on mech passed params to decode.